Sources/GRPCHTTP2Core/Client/Connection/LoadBalancers/Subchannel.swift

```diff
@@ -329,22 +329,21 @@ extension Subchannel {
     guard isClosed else { return }
 
     switch reason {
-    case .idleTimeout:
-      // Connection closed due to an idle timeout; notify the load balancer about this.
+    case .idleTimeout, .remote, .error(_, wasIdle: true):
+      // Connection closed due to an idle timeout or the remote telling it to GOAWAY; notify the
+      // load balancer about this.
```

Comment:
For the `.error(_, true)` case: I understand that if there were no open streams, then the connection was idle. However, is it correct to say that the connection was closed due to an idle timeout? Couldn't we have closed for some other unrelated error and it just so it happened to be that there were no open streams?

Reply:
> However, is it correct to say that the connection was closed due to an idle timeout? 

It isn't correct, no. 

However, we aren't saying that. The underlying connection is telling us why it closed and we're saying that as a result of that the state of the subchannel has changed to idle (i.e. there's no connection and we aren't trying to create one).